qa/tasks/ceph_manager.py

```diff
@@ -453,8 +454,10 @@ def kill_osd(self, osd=None, mark_down=False, mark_out=False):
                                '--journal-path', JPATH.format(id=imp_osd),
                            ])
                            + " --op apply-layout-settings --pool " + pool).format(id=osd)
-                    output = imp_remote.sh(cmd, wait=True, check_status=False)
-                    if 'Couldn\'t find pool' in output:
+                    proc = imp_remote.run(args=cmd,
+                                          wait=True, check_status=False,
+                                          stderr=BytesIO)
+                    if 'Couldn\'t find pool' in six.ensure_str(proc.stderr.getvalue()):
```

Comment:
If we want here to process stderr, I would prefer to do:
```
errors=BytesIO()
output = imp_remote.sh(cmd, wait=True, stderr=errors, check_status=False)
if b'Couldn\'t find pool' in errors.getvalue():
    . . .
```
```

Reply:
@kshtsk, could you take a look at the commit message of https://github.com/ceph/ceph/pull/33805/commits/0b30d99424d2a2252d3e29a71cb6fb3a21891fc1 ? does it make sense to you?

Reply:
I still don't understand why BytesIO can not be used here? The command output by default is bytes, using StringIO is confusing for py2/py3 compatibility.

Reply:
i think i've explain the rationale in the commit message. it might be related to py2/py3. but it's more related to what the consumers of this method expect from the output.

Reply:
I might be wrong but consumer expects bytes all the time uses run, since originally the remote.run was using cString.StringIO which is io.BytesIO in python2.

Reply:
right. because python2 does not differentiate byte from str, so the expectation back then is not obvious if we look at the type of the the writeable object passed to `run()` -- even if the consumers expects `str`, they still pass `StringIO`, and apparently they are happy with it, because python2 works this way.

but in python3, we have to make a decision. in this case, i think `StringIO` is better, as we don't expect binary here. and actually, we could avoid setting `ChannelFile.FLAG_BINARY` flag when collecting the output from COT, and consider it a bug/failure if COT emits non-ASCII output when teuthology is noting its output down with `logging`.

Reply:
My complain is that there a confusion between io.StringIO and six.StringIO and their incompatibility among py2 and py3, this is why I prefer io.BytesIO, which is always bytes in py2 and py3, and use six.ensure_str if want to process output as str.

Reply:
Btw,
```
                                          stderr=BytesIO)
```
That was unfortunate typo, it should be 'BytesIO()'.

Reply:
thanks for pointing this out, you are looking at a snapshot of the change. it's has been fixed.

Reply:
actually, my focus is python3. after moving teuthology over to python3, we need to ditch `six` in master.